state-sync/src/coordinator.rs

```diff
@@ -1325,22 +1380,34 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             );
         }
 
-        // verify the end-of-epoch LI for the following before passing it to execution:
-        // * verify end-of-epoch-li against local state
-        // * verify end-of-epoch-li's version corresponds to end-of-chunk version before
-        // passing it to execution
-        // * Executor expects that when it is passed an end-of-epoch LI, it is going to execute/commit
-        // transactions leading up to that end-of-epoch LI
-        // * verify end-of-epoch-li actually ends an epoch
-        let end_of_epoch_li = end_of_epoch_li
-            .map(|li| self.local_state.verify_ledger_info(&li).map(|_| li))
-            .transpose()?
-            .filter(|li| {
-                li.ledger_info().version() == known_version && li.ledger_info().ends_epoch()
-            });
+        // Verify the end_of_epoch_li against local state and ensure the version
+        // corresponds to the version at the end of the chunk.
+        // The executor expects that when it is passed an end_of_epoch_li, it is going
+        // to execute/commit transactions leading up to that li, so we
+        // also verify that the end_of_epoch_li actually ends the epoch.
+        let mut verified_end_of_epoch_li = None;
+        if let Some(end_of_epoch_li) = end_of_epoch_li {
```

Comment:
set the verified_end_of_epoch_li here avoid making mut variables if at all possible

Reply:
Generally, I agree, but I also tend to avoid it when I'd have 3 branches and 2 of those branches return the same thing (e.g., None). Personally, I find only 2 branches easier to reason about, but I've changed it for you 😄